backend/src/components/user.js

```diff
@@ -225,7 +230,57 @@ function mapUserObjectForBack(data) {
 async function createUser(req, res) {
   try {
     const payload = mapUserObjectForBack(req.body)
-    let response = await postOperation('contacts', JSON.stringify(payload))
+    let response = await postOperation('contacts', payload)
+    let returnVal = new MappableObjectForFront(response, UserMappings).toJSON()
+    returnVal.role = Number(returnVal.role)
+    return res.status(HttpStatus.OK).json(returnVal)
+  } catch (e) {
+    return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json(e.data ? e.data : e?.status)
+  }
+}
+
+function mapContactForBack(data) {
+  let contact = new MappableObjectForBack(data, ContactMappings).toJSON()
+  contact.entityNameSet = 'contacts'
+  contact.actionMode = 'Update'
+  return contact
+}
+
+function mapContactFacilitiesForBack(data) {
+  let facilitiesForBack = []
+  data.facilities.forEach((facilityFromFront) => {
+    let facility = {}
+    facility.entityID = facilityFromFront.bceidFacilityId
+    facility.ofm_portal_access = facilityFromFront.ofmPortalAccess
+    facility.entityNameSet = 'ofm_bceid_facilities'
+    facility.actionMode = 'Update'
+    facilitiesForBack.push(facility)
+  })
+  return facilitiesForBack
+}
+
+async function updateUser(req, res) {
+  try {
+    let payload = {
+      batchTypeId: 101,
+      feature: 'AccountManagement',
+      function: 'UserEdit',
+      actionMode: 'Update',
+      scope: 'Parent-Child',
```

Comment:
In the future if we're seeing these values used in multiple places we may want to make constants for some of these.

Reply:
Understood. Left as is.